webpack/ForemanTasks/Components/TasksTable/TasksTableReducer.js

```diff
@@ -0,0 +1,50 @@
+import Immutable from 'seamless-immutable';
+import { combineReducers } from 'redux';
+import { createTableReducer } from 'foremanReact/components/common/table';
+import {
+  TASKS_CONTROLLER,
+  TASKS_REQUEST,
+  TASKS_SUCCESS,
+  TASKS_FAILURE,
+} from './TasksTableConstants';
+import { FOREMAN_TASKS_DASHBOARD_UPDATE_QUERY } from '../TasksDashboard/TasksDashboardConstants';
+
+const initialState = Immutable({
+  query: null,
+  loading: true,
+  itemCount: 0,
+  pagination: {
+    page: 1,
+    perPage: 20,
+  },
+});
+
+export const TasksTableQueryReducer = (state = initialState, action) => {
+  const { type, payload } = action;
+  switch (type) {
+    case TASKS_REQUEST:
+      return state.set('loading', true);
+    case TASKS_FAILURE:
+      return state.set('loading', false);
+    case TASKS_SUCCESS:
+      return Immutable.merge(state, {
+        loading: false,
+        itemCount: payload.subtotal,
+        pagination: {
+          page: parseInt(payload.page, 10),
+          perPage: parseInt(payload.per_page, 10),
+        },
+      });
+    case FOREMAN_TASKS_DASHBOARD_UPDATE_QUERY:
+      if (payload.mode === 'last') {
```

Comment:
It doesn't feel right here, can't we store the data as is? or pass a payload with the right format?